docs/reference/builder.md

```diff
@@ -152,18 +152,22 @@ Example (parsed representation is displayed after the `#`):
     ADD . $foo       # ADD . /bar
     COPY \$foo /quux # COPY $foo /quux
 
-The instructions that handle environment variables in the `Dockerfile` are:
+Environment variables are supported by the following list of instructions in 
+the `Dockerfile`:
 
-* `ENV`
 * `ADD`
 * `COPY`
-* `WORKDIR`
+* `ENV`
 * `EXPOSE`
-* `VOLUME`
+* `ONBUILD` (see note)
```

Comment:
I'd suggest replacing the `(see note)` with something that indicates that only ONBUILD instructions that use one of the other listed instructions support the use of `$env`